derive_builder_core/src/build_method.rs

```diff
@@ -63,12 +65,21 @@ impl<'a> ToTokens for BuildMethod<'a> {
             BuilderPattern::Immutable => quote!(&self),
         };
         let doc_comment = &self.doc_comment;
+
+        let (result, string) = if self.no_std {(
+            quote!(::core::result::Result),
+            quote!(::collections::string::String),
+        )} else {(
+            quote!(::std::result::Result),
+            quote!(::std::string::String),
+        )};
+
         if self.enabled {
             trace!("Deriving build method `{}`.", self.ident.as_ref());
             tokens.append(quote!(
                 #doc_comment
                 #vis fn #ident(#self_param)
-                    -> ::std::result::Result<#target_ty #target_ty_generics, ::std::string::String>
+                    -> #result<#target_ty #target_ty_generics, #string>
```

Comment:
This is the reason `extern crate collections` is required, right? Can we work around that? E.g., not give string error messages when using `no_std` but more simple ones?